sycl/include/CL/sycl/INTEL/online_compiler.hpp

```diff
@@ -0,0 +1,192 @@
+
+  /// Sets the device type of the compilation target and returns *this.
+  online_compiler<Lang> &setTargetDeviceType(sycl::info::device_type type);
+
+  /// Sets the device architecture of the compilation target and returns *this.
+  online_compiler<Lang> &setTargetDeviceArch(device_arch arch);
+
+  /// Makes the compilation target 32-bit and returns *this.
+  online_compiler<Lang> &set32bitTarget();
+
+  /// Makes the compilation target 64-bit and returns *this.
+  online_compiler<Lang> &set64bitTarget();
+
+  /// Sets implementation-defined target device stepping of the compilation
+  /// target and returns *this.
+  online_compiler<Lang> &setTargetDeviceStepping(const std::string &id);
+
+private:
+  // Compilation target specification fields: {
+
+  /// Compiled code format.
+  compiled_code_format OutputFormat;
+
+  /// Compiled code format version - a pair of "major" and "minor" components
+  std::pair<int, int> OutputFormatVersion;
+
+  /// Target device type
+  sycl::info::device_type DeviceType;
+
+  /// Target device architecture
+  device_arch DeviceArch;
+
+  /// Whether the target device architecture is 64-bit
+  bool Is64Bit;
+
+  /// Target device stepping (implementation defined)
+  std::string DeviceStepping;
+  // }
+};
+
+// Specializations of the online_compiler class and 'compile' function for
+// particular languages and parameter types.
+
+/// Compiles given OpenCL source. May throw \c online_compile_error.
+/// @param src - contents of the source
+template <>
+template <>
+std::vector<byte> online_compiler<source_language::OpenCL_C>::compile(
+  const std::string &src) {
+  // real implementation will call some non-templated impl function here
+  return std::vector<byte>{};
+}
+
+/// Compiles given OpenCL source. May throw \c online_compile_error.
+/// @param src - contents of the source
+/// @param options - compilation options (implementation defined); standard
+///   OpenCL JIT compiler options must be supported
+template <>
+template <>
+std::vector<byte> online_compiler<source_language::OpenCL_C>::compile(
+    const std::string &src, const std::vector<std::string> &options) {
+  // real implementation will call some non-templated impl function here
+  return std::vector<byte>{};
+}
+
+/// Compiles given CM source.
+template <>
+template <>
+std::vector<byte> online_compiler<source_language::CM>::compile(
+    const std::string &src) {
+  // real implementation will call some non-templated impl function here
+  return std::vector<byte>{};
+}
+
+/// Compiles given CM source.
+/// @param options - compilation options (implementation defined)
+template <>
+template <>
+std::vector<byte> online_compiler<source_language::CM>::compile(
+    const std::string &src, const std::vector<std::string> &options) {
+  // real implementation will call some non-templated impl function here
+  return std::vector<byte>{};
+}
+
+} // namespace sycl
+} // namespace INTEL
+} // __SYCL_INLINE_NAMESPACE(cl)
```

Comment:
Please add another line after this one
```suggestion
} // __SYCL_INLINE_NAMESPACE(cl)

```